crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -0,0 +1,190 @@
+    let _b_nonnormalized_dagger = Scale(c64 { re: 0.5, im: 0.0 }) * _b_nonnormalized.adjoint();
+    _b_nonnormalized_dagger * unitary * _b_nonnormalized
+}
+
+// fn transform_to_magic_basis(unitary : Mat<c64>) -> Mat<c64> {
+//     let _b_nonnormalized : Mat<c64> = mat![[C1, C1IM, C0, C0],
+//                                            [C0, C0, C1IM, C1],
+//                                            [C0, C0, C1IM, -C1],
+//                                            [C1, -C1IM, C0, C0]];
+//     let _b_nonnormalized_dagger = Scale(c64 {re: 0.5, im: 0.0}) * _b_nonnormalized.adjoint();
+//     _b_nonnormalized * unitary * _b_nonnormalized_dagger
+// }
+
+// faer::c64 and num_complex::Complex<f64> are both structs
+// holding two f64's. But several functions are not defined for
+// c64. So we implement them here. These things should be contribute
+// upstream.
+
+pub trait PowF {
+    fn powf(self, pow: f64) -> c64;
+}
+
+impl PowF for c64 {
+    fn powf(self, pow: f64) -> c64 {
+        c64::from(Complex::<f64>::from(self).powf(pow))
+    }
+}
+
+pub trait Arg {
+    fn arg(self) -> f64;
+}
+
+impl Arg for c64 {
+    fn arg(self) -> f64 {
+        let c = Complex::<f64>::from(self);
+        c.arg()
+    }
+}
+
+fn __weyl_coordinates(unitary: MatRef<c64>) -> (f64, f64, f64) {
+    let pi = PI;
+    let pi2 = PI / 2.0;
+    let pi4 = PI / 4.0;
+    let uscaled = Scale(C1 / unitary.determinant().powf(0.25)) * unitary;
+    let uup = transform_from_magic_basis(uscaled);
+    let mut darg: Vec<_> = (uup.transpose() * &uup)
+        .complex_eigenvalues()
+        .into_iter()
+        .map(|x: c64| -x.arg() / 2.0)
+        .collect();
+    darg[3] = -darg[0] - darg[1] - darg[2];
+    let mut cs: Vec<_> = (0..3)
+        .map(|i| utils::modulo((darg[i] + darg[3]) / 2.0, 2.0 * pi))
+        .collect();
+    let cstemp: Vec<f64> = cs
+        .iter()
+        .map(|x| utils::modulo(*x, pi2))
+        .map(|x| x.min(pi2 - x))
+        .collect();
+    let mut order = utils::argsort(&cstemp);
+    (order[0], order[1], order[2]) = (order[1], order[2], order[0]);
+    (cs[0], cs[1], cs[2]) = (cs[order[0]], cs[order[1]], cs[order[2]]);
+
+    // Flip into Weyl chamber
+    let pi32 = 3.0 * pi2;
+    if cs[0] > pi2 {
+        cs[0] -= pi32;
+    }
+    if cs[1] > pi2 {
+        cs[1] -= pi32;
+    }
+    let mut conjs = 0;
+    if cs[0] > pi4 {
+        cs[0] = pi2 - cs[0];
+        conjs += 1;
+    }
+    if cs[1] > pi4 {
+        cs[1] = pi2 - cs[1];
+        conjs += 1;
+    }
+    if cs[2] > pi2 {
+        cs[2] -= pi32;
+    }
+    if conjs == 1 {
+        cs[2] = pi2 - cs[2];
+    }
+    if cs[2] > pi4 {
+        cs[2] -= pi2;
+    }
+    (cs[1], cs[0], cs[2])
+}
+
+// For debugging. We can remove this later
+#[pyfunction]
+#[pyo3(text_signature = "(unitary, /")]
+pub fn _weyl_coordinates(unitary: PyReadonlyArray2<Complex<f64>>) -> (f64, f64, f64) {
+    let u = unitary.as_array().into_faer_complex();
+    __weyl_coordinates(u)
+}
+
+#[pyfunction]
```

Comment:
Do you mind adding a docstring here just so that `__doc__` in python is exposing something. We won't publish this doc, but it can be useful for people that have IDEs that expose `__doc__`. Something like:

```suggestion
/// Compute the number of basis gates needed in a decomposition of the input unitary
#[pyfunction]
```

Reply:
Good idea. But better to remove the function.

Reply:
No need for docstring, function removed in 96a3f80